### DISCUSSION TITLE
How to get response headers?

### DISCUSSION THREAD

**Discussion by revnelson** [jycouet/kitql#151] [Q&A] [answered]
I am building a project for headless WordPress and I need to get a cart session token that is sent in the response headers from the API. I tried adding `if (res.headers.get('woocommerce-session')) {  dataToReturn.session = res.headers..get('woocommerce-session') }` around line 365 in `@kitql/client/index.js`. Logging the res in `queryLocal` of a generated store shows the session token is correctly returned in the server console. However, in the browser console the same log doesn't even have a session key. If I add `{JSON.stringify($KQL_GetCart, null, 2)}` to a page, it looks like the session token flashes for a second before being removed. I can't seem to find where the shape of the store is being enforced. Unfortunately, I need the token on the client side to store in LocalStorage or something similar to recall cart data later.

**Comment by jycouet** [comment]
Yes, this is not sooo easy today. 
But I'll have a big update soon where it will be easier to do.

Today, you can do it with `cookie`, you can maybe have a look at: https://github.com/notrab/graphql.wtf-kitql

**Comment by jycouet** [comment, marked as answer]
Check out the [Announcement here](https://github.com/jycouet/kitql/discussions/156#discussioncomment-3080343)

---

Now you can hook up before or after the fetch, 

Here is an example 👇
```ts
import type { RequestHandlerArgs } from '$houdini';
import { HoudiniClient } from '$houdini';
import { stry } from '@kitql/helper';

// For Query & Mutation
async function fetchQuery({
  fetch,
  text = '',
  variables = {},
  session,
  metadata
}: RequestHandlerArgs) {
  // Prepare the request
  const url = import.meta.env.VITE_GRAPHQL_ENDPOINT || 'http://localhost:4000/graphql';

  // regular fetch (Server & Client)
  const result = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${session?.token}` // session usage example
    },
    body: JSON.stringify({
      query: text,
      variables
    })
  });

  // return the result as a JSON object to Houdini
  const json = await result.json();

  // metadata usage example
  if (metadata?.logResult === true) {
    console.info(stry(json, 0));
  }

  return json;
}

// Export the Houdini client
export const houdiniClient = new HoudiniClient(fetchQuery);

```
